src/QMCDrivers/MoveContext.h

```diff
@@ -0,0 +1,97 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2019 developers.
+//
+// File developed by: Peter Doak, doakpw@ornl.gov, Oak Ridge National Laboratory
+//
+// File refactored from ParticleSet.h
+//////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef QMCPLUSPLUS_MOVECONTEXT_H
+#define QMCPLUSPLUS_MOVECONTEXT_H
+
+#include <map>
+#include <vector>
+#include "OhmmsSoA/VectorSoaContainer.h"
+#include "Configuration.h"
+#include "Particle/Walker.h"
+#include "QMCDrivers/Crowd.h"
+#include "ParticleBase/RandomSeqGenerator.h"
+
+namespace qmcplusplus
+{
+
+class MCPopulation;
+class DistanceTableData;
+/** Thread local context for moving walkers
+ *
+ *  created on the stack once per driver per crowd
+ *  refactored out of ParticleSet, only the minimum
+ */
+class MoveContext
```

Comment:
I read 'Move' as a verb (even in the class name) and I expect this to have to do with moving data in the program (vs. moving walker positions in space).
ParticleMoveContext?  CrowdParticleMoveContext?  DanceContext? (when a bunch of particles move, we could call it a dance? :smiley: That's probably getting too far outside standard domain language, though.)

Reply:
That hadn't occurred to me but now that's how I read it too. Will rename.